packages/browser/examples/demoClientApp/src/DemoClientApp.js

```diff
@@ -296,31 +576,10 @@ class DemoClientApp extends Component {
               </span>
               {this.highlightPartOfWebId(this.state.sessionInfo.webId, 2)}
             </p>
-            <form>
-              <div style={style}>
-                <input
-                  size="80"
-                  type="text"
-                  value={this.state.fetchRoute}
-                  onChange={(e) =>
-                    this.setState({ fetchRoute: e.target.value })
-                  }
-                />
-                <button onClick={this.handleFetch}>Fetch</button>
-              </div>
-              <p></p>
-              <div style={style}>
-                <strong>Resource:</strong>
-                <pre>{this.state.fetchBody}</pre>
-              </div>
-            </form>
 
-            <p></p>
-            <div>
-              <form>
-                <button onClick={this.handleLogout}>Log Out</button>
-              </form>
-            </div>
+            {this.htmlFetchResource()}
+
+            {this.htmlLogout()}
           </div>
         );
     }
```

Comment:
This file is starting to get huge : at some point, this should be split into smaller components, and be architectured according to React good practices. e.g. changing the class components into functional ones and hooks.

Reply:
Yeah, still lots more tidying up to be done - this would be just part of all that.